### PR TITLE
Pass on query strings/paths from referrer, keeping reference to origin with 'optedvia' string

### PR DESCRIPTION
**Pull request by keirog**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -4,18 +4,26 @@
...
 var app = express();
 var cookieParser = require('cookie-parser');
+var url = require('url');
+var assign = require('lodash.assign');
 var errorMiddleware = require('express-errors-handler').middleware;
 var metrics = require('next-metrics');
 var debug = require('debug')('next-opt-in-out-debug');
 
-var port = Number(process.env.PORT || 5003);
+var PORT = Number(process.env.PORT || 5003);
```
Comment (aintgoin2goa) [inline app.js]: why Number constructor?  I doubt you need to worry about casting this to a number but if you do surely parseInt is better?

File: app.js [hunk 2]
```diff
@@ -23,26 +31,42 @@ app.get('/*', function(req, res, next){
 
 app.get('/opt-in', function(req, res){
 	debug('Opt-In');
-	// See /temp-opt-out
-	res.clearCookie('FT_TEMP_SITE', { path: 'ft.com' });
 
-	res.cookie('FT_SITE', 'NEXT', { maxAge: 2592000000, domain:'ft.com' });
+	var newUrl = url.parse(URL_NEXT, true);
+	var referrer = req.get('Referer');
```
Comment (aintgoin2goa) [inline app.js]: Can we trust the Referrer header?
http://en.wikipedia.org/wiki/HTTP_referer#Referer_hiding

Should we just pass something in the query string?

File: app.js (hunk 1 above)
Comment (keirog) [inline app.js]: No strong opinion either way on this one; both have fors and againsts. Happy to change

**Comment by aintgoin2goa** [comment]
:+1:

**Pull request merged**